### DISCUSSION TITLE
Tanstack router typed routes

### DISCUSSION THREAD

**Discussion by andredewaard** [vuejs/router#2368] [Ideas]
I know you guys are working on a typed route implementation using this https://router.vuejs.org/guide/advanced/typed-routes which is already under development for some time now.

but i really love the Tanstack Router method for typed routes. Is this something you're looking at for inspiration?

**Comment by posva** [comment]
We didn't look at Tanstack Router for inspiration (it didn't even exist back when I started typed routes). What do you miss here?
The thing that I would like to add in a major is a big refactor to the `<RouterLink>` component to adat better to types.
I am also working on other things like custom parsing / serializing of route params and other _matcher_ refactorings
